### PR TITLE
[614] Port over course requirements flow

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/h6asrgSp/614-migrate-courses-course-show-page-edit-requirements-eligibility-qualities-requirements

### Changes proposed in this pull request

- Port over the course requirement editing flow

### Guidance to review

Original implementation here: https://qa.publish-teacher-training-courses.service.gov.uk/organisations/2AT/2022/courses/AK12/requirements#personal-qualities

migrated version: https://teacher-training-api-pr-2544.london.cloudapps.digital/publish/organisations/2AT/2022/courses/AK12/requirements#personal-qualities

Check:

You can update the course requirements

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Migrate Courses - Course show page - edit requirements & eligibility (qualities & requirements)](https://trello.com/c/h6asrgSp/614-migrate-courses-course-show-page-edit-requirements-eligibility-qualities-requirements)

File: app/views/publish/courses/requirements/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    ) do |f| %>
+
+      <%= f.govuk_error_summary %>
+
+      <h1 class="govuk-heading-l">
+        <span class="govuk-caption-l"><%= @course.name_and_code %></span>
+        <%= page_title %>
+      </h1>
+
+      <% if @provider.recruitment_cycle_year.to_i < Provider::CHANGES_INTRODUCED_IN_2022_CYCLE %>
```
Comment (darokel) [inline app/views/publish/courses/requirements/edit.html.erb:30]: @defong Do we still need to deal with this?

File: app/forms/publish/course_requirement_form.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    validates :personal_qualities, words_count: { maximum: 100,  message: :too_long }
+    validates :other_requirements, words_count: { maximum: 100,  message: :too_long }
+
+  private
+
+    def compute_fields
+      course_enrichment.attributes.symbolize_keys.slice(*FIELDS).merge(new_attributes)
+    end
+
+    def required_qualifications_needed?
```
Comment (darokel) [inline app/forms/publish/course_requirement_form.rb:24]: Is this redundant now that we're in this cycle at the moment?

File: app/views/publish/courses/requirements/edit.html.erb (hunk 1 above)
Comment (defong) [reply app/views/publish/courses/requirements/edit.html.erb:30]: That can be drop, tech debt that was incurred due to last rollover changes

File: app/forms/publish/course_requirement_form.rb (hunk 2 above)
Comment (defong) [reply app/forms/publish/course_requirement_form.rb:24]: Yes

**Review by defong — APPROVED** [review]

**Pull request merged**